### ISSUE TITLE
When I launch $gulp test:auto and $gulp serve tests are compiled into application

### ISSUE THREAD

**Issue by TomaszScisloGS** [Swiip/generator-gulp-angular#928]
When I do:
1. $gulp test:auto and $gulp server in parallel 
I get error in JS console from which I get the impression that the tests are compiled with the TS source code.

**Comment by Swiip** [comment]
That looks like an old bug, which version do you have?

**Comment by TomaszScisloGS** [comment]
generator-gulp-angular 1.0.0

**Comment by Swiip** [comment]
Arf, no, I read to quiclky. It's not this old bug.

Tests compile sources adding specs inside the bundle, serve compile without them. But both put the generated bundle at the same place. One will override the other then.

You can just ignore this as it's not a problem you'll have when you don't launch the tests in parallel. Or you have to try change the configuration of the test process to use a bundle with another name like bundle-test.js. But I fear you have to change many configurations to make it working.

If you succeed and are very brave, I'm interested in a PR for this :)

**Comment by vlio20** [comment]
@Swiip I am encountering this: test and then build then my test code is present in my dist scripts. Shouldn't the clean task be a dependency of the build (the very first thing to run)?

**Comment by tscislo** [comment]
@vlio20 try launching default task. It will clean dist and then it will hit build.

**Comment by vlio20** [comment]
@tscislo, thanks it's great that there is a way to build cuttectly with the default task but the build task not works as expected

**Comment by Swiip** [comment]
build should trigger scripts task which will recompile and override the bundle without the tests. Can you check if or why this is not happening?

**Comment by TomaszScisloGS** [comment]
@Swiip 
I also encountered this error which @vlio20 mentioned.
Scenario is like this:
$ gulp
$ gulp test

And you have test suites compiled into source code. Sometimes above is the order which you need. For example using your CI tools. Right now only the following sequence works:
$gulp
$gulp test
$gulp

Ideas?

**Comment by Swiip** [comment]
The commands you give are launched sequentially? If yes, I don't see how the compiled bundle could be impacted by something happening after. If it's in parallel, it's simple, both tasks gulp and gulp test use the same temporary file .tmp/serve/index.module.js and one will override the other.

Perhaps the easiest for you is to create a gulp ci test which sequence the other two.

**Comment by TomaszScisloGS** [comment]
@Swiip you're right I had a misconception of gulp tasks not being run in parallel, so my last comment regarding tests and build is not valid any more. However the first one - the ability to run in parallel gulp test:auto and gulp serve is still valid. I think that this is a very important functionality to have.

**Comment by Swiip** [comment]
Yep, it's a feature to have. I think the real good solution is to move webpack inside karma with the karma webpack plugin. I tested it in a private project. It's faster and don't have this issue. It also enables the possibility to have ES6 coverage with isparta...

But it's a lot of ifs in the karma.conf.js whether you use webpack (ES6 / TS) or not (pure JS).
